app/distros/octane.js

```diff
@@ -0,0 +1,79 @@
+var request = require('request');
+var merge = require('merge');
+
+function init() {
+  var module = {
+    id: 'octane',
+    // Generated project will be based on Lightning, not Octane directly.
+    // Octane only used as template for build files.
+    profile: 'lightning'
+  };
+
+  module.option = {
+    name: 'Octane/Lightning',
+    value: module.id
+  };
+
+  module.versions = [
+    {name: 'Octane 8.x', value: '8.x'}
+  ];
+
+  module.versionDefault = '8.x';
+
+  module.description = 'This project is built on [' + module.option.name + '] distribution.';
+
+  module.releaseVersion = function(majorVersion, done, cb) {
+    cb(null, '8.x', done);
+  };
+
+  module.drushMakeFile = function(yo, options, done) {
+    // Using all composer.  No Drush make files.
+    done();
+  };
+
+  module.modifyComposer = function(yo, options, composer, isNewProject, done) {
+    // We fetch the current composer.json from Drupal.org for the distro.
+    // Then we merge the "require", "require-dev", patches" sections with either
+    // the existing project composer.json, or the Drupal base template.
+
+    // Lastest Octane distro code living on drupal.org
+    var url = 'http://cgit.drupalcode.org/' + options.drupalDistro.id
```

Comment:
if (options.offline) { response should be an empty object; } else { make the request }